diff -up firefox-75.0/intl/locale/LocaleService.cpp.locale-debug firefox-75.0/intl/locale/LocaleService.cpp
--- firefox-75.0/intl/locale/LocaleService.cpp.locale-debug	2020-03-10 11:09:25.839894237 +0100
+++ firefox-75.0/intl/locale/LocaleService.cpp	2020-03-10 11:09:59.824675890 +0100
@@ -371,9 +371,7 @@ LocaleService::GetDefaultLocale(nsACStri
     // update.locale file is not found, item.len will remain 0 and we'll
     // just use our hard-coded default below.
     GetGREFileContents("update.locale", &locale);
-    locale.Trim(" \t\n\r");
-    // This should never be empty.
-    MOZ_ASSERT(!locale.IsEmpty());
+    locale.Trim(" \t\n\r");    
     if (CanonicalizeLanguageId(locale)) {
       mDefaultLocale.Assign(locale);
     }
